### ISSUE TITLE
Query about your mobx-tree models

### ISSUE THREAD

**Issue by airtonix** [ProteanDev/react-phaser3-mobX-template#1]
Thanks for making this repository available, It's opened some horizons my Typescript learnings.

I have some questions about what your intentions for various pieces your implementation of the models. 

I get the feeling that this is designed to work with a boardgame or roguelike game where you control many pieces?

### Model Relationship

 You have `Character`, `Actor`, `Player` and `Entity`.

#### `types.reference(CharacteMST)` vs `types.array(types.string)`

- How come you only store the Character id under the `player.characters[]` ? I can assume for performance.
- If we're only storing Ids of characters where do we access the CharacterModel instance later to access its properties and actions? https://github.com/ProteanDev/react-phaser3-mobX-template/blob/master/src/core/actions/getPlayerActions.ts#L5-L8.

Looking at the tutorials of MST, perhaps I can assume that you bring all these models together with some kind of `Game` model as a Root Store?

#### Difference between `Player`, `Character`, `Actor`, `Entity`

- `Character`: piece controlled by the player?
- `Player`: A container for Characters
- `Actor`: A non player character, monster, chest, shopkeeper?
- `Entity`: items found in chests, players inventory, on the ground, looted from `Actor`s, traded from `Actor`s?

**Comment by ProteanDev** [comment]
Yeah about that, that part is still a draft so feel free to ignore it for now, but thanks for mentioning. When I get some free time, I'll be working on this again.
